
Cancer Systems Biology, Section of Bioinformatics, Department of Health and Technology, Technical University of Denmark, 2800, Lyngby, Copenhagen

# Moonlight2_DMA_basal_like


An Automatized Workflow to Study Mechanistic Indicators for Driver Gene Prediction with Moonlight
Astrid Saksager, Mona Nourbakhsh, Nikola Tom, Xi Steven Chen, Antonio Colaprico, Catharina Olsen, Matteo Tiberti, Elena Papaleo*, submitted to biorxiv


contacts for the repository: Elena Papaleo, elpap-at-dtu.dk, elenap-at-cancer.dk; Astrid Saksager: 


PLEASE, CITE THE PUBLICATION ABOVE IF YOU USE THE SCRIPTS FOR YOUR OWN RESEARCH

This Project-folder contains the entire Basal-like Breast Cancer study
AIM: To find driver genes for Basal-like patients by running Moonlight on BRCA-Basal specific data from TCGA.

## SET UP 

#### CONDA:
##### install:
conda create --prefix ./env_Basal -c conda-forge r-base=4.1.3 r-devtools
##### to activate
conda activate ./env_Basal

## General information

The scripts in /scripts/ and data in /data/raw_data/ is the only things needed to run this entire projct.
There were thrown error messages when GRN where run with tsp. The solution where to:
$ Xvfb :88 -screen 0 1920x1080x16 &
$ diswown %1                            #change 1 to number displayed in brackets from above
$ export display=:88

$ ./Run_Basal.sh                        # Without tsp in terminal

## FOLDERS:

### /data/
The data generated by moonlight:
Basal_FEA.rda, Basal_URA.rda, Basal_GRN.rda, Basal_PRA.rda, Basal_DMA.rda

### /rawdata/
The data that is not generated in this project. It was processed somewhere else on the server
They are all symbolic links
* Mutation from TCGA-BRCA
* Differential Gene expression from TCGA-BRCA
* Normal counts of gene expression

### /scripts/
The script 00_init.R loads and checks the prirequirements to run this project.
All scripts are then run in the correct order.
The script are numbered and the results from each are numbered accordingly.

### /results/
The three additional tables of data from DMA (run_moonlight.R)
Plots and tables used in the paper
The numbers refere to the script they are generated from

### PAPER CONTENTS 


Figure 1: Not in repository - created in biorender
Figure 2: 021_heatmap_complete.pdf
Figure 3: 022_Mediators_moonlight_heatmap.pdf
Figure 4: 032_promoter_moonlight_heatmap.pdf
Figure 5: 05_GO_KEGG_top_adjpval_OCG_TSG.png
Figure 6: 06_upsetplot_drivers.pdf

Table 1: The CScape scores are derived from Rogers et al. 2020, see  website or original cscape scripts
Table 2: The summarised numbers are calculated in scripts/030_summarise.R
Table 3: 032_promoter_mutation_genes.csv
Table 4: 033_driver_TF.csv
Table S3: 06_overlap_moonlight_drivers_vs_ncg.csv
Table S4: 07_overlapping_genes_moonlight_ncg_gust.csv

Supplementary Figures:
Sub figure 1: 05_variant_class_percent_driver_TSG_OCG.png
Sub figure 2: 98_mutations_per_patient.png
